src/libYARP_OS/include/yarp/os/FixedSizeBuffersManager.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_OS_FIXEDSIZEBUFFERSMANAGER_H
+#define YARP_OS_FIXEDSIZEBUFFERSMANAGER_H
+
+#include <yarp/os/Mutex.h>
+#include <vector>
+#include <stdexcept>
+
+
+// Defined in this file:
+namespace yarp { namespace os { template <typename T> class FixedSizeBuffersManager; }}
+namespace yarp { namespace os { template <typename T> class Buffer; }}
+
+
+namespace yarp {
+namespace os {
+
+
+/**
+* Buffer contains info about a buffer of type T and it is used to exchange information
+* with yarp::os::FixedSizeBuffersManager.
+*/
+
+template <typename T>
+class Buffer
+{
+friend class FixedSizeBuffersManager<T>;
+
+private:
+    uint32_t key;
+    T* dataPtr;
```

Comment:
I don't know how is feasible but for new classes I discourage the usage of C pointers, I'd prefer `std::unique_ptr`.
`unique_ptr` has overhead only in constructor and destructor if they are not trivial, but it is not the case if `T` is a literal type.

Reply:
I think `std::unique_ptr` is a good feature that let developer to forget `delete` statements, but in this case I prefer continue to use pointer C style, because this is a Manager of buffer and its abojective is alloc and dealloc memory in the right moment. 

Moreover, I already tested the buffer and it works.
So I don't see the need to change something to seems ok without add real improvements.

The  `std::unique_ptr` can be use in user-code where it is necessary to allocate and to deallocate memory most frequently, in order to avoid the usual bug about delete (forgot a delete or delete something not exists anymore)

Reply:
I think the `Buffer` does not own the memory pointed by `dataPtr` (that is actually owned by `FixedSizeBuffersManager`, and "shared" with the Buffer object), so I do not think that `unique_ptr` can be used here.